File: apple/Sources/ReactNativeWatchOS/Modifiers/LabelStyle.swift
import SwiftUI

/// SwiftUI `.labelStyle(_:)`. Maps the `style` string to a label style.
/// Unknown values fall back to `.automatic`.
enum RNWLabelStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("labelStyle") { view, params, _ in
            switch params.string("style") {
            case "iconOnly":
                return AnyView(view.labelStyle(.iconOnly))
            case "titleOnly":
                return AnyView(view.labelStyle(.titleOnly))
            case "titleAndIcon":
                return AnyView(view.labelStyle(.titleAndIcon))
            default:
                return AnyView(view.labelStyle(.automatic))
            }
        }
    }
}

File: src/modifiers/labelStyle.ts
import { createModifier } from './createModifier';

export type LabelStyle =
  | 'automatic'
  | 'iconOnly'
  | 'titleOnly'
  | 'titleAndIcon';

/// SwiftUI `.labelStyle(_:)`. Sets the visual style applied to `Label`s
/// within the view. `'automatic'` defers to the platform default.
export function labelStyle(style: LabelStyle = 'automatic') {
  return createModifier('labelStyle', { style });
}
